src/v1/waittimes/routes.ts

```diff
@@ -0,0 +1,46 @@
+import * as express from 'express';
+import { Redis } from 'ioredis';
+import { WaitTimeDB } from './db';
+import asyncify from '../lib/asyncify';
+import { cache } from '../lib/cache';
+import { generateKey } from '../server';
+
+import bodyParser = require("body-parser");
+
+export default function routes(redis?: Redis, credentials?) {
+  const db = new WaitTimeDB();
+
+  const router = express.Router();
+  router.use(bodyParser.json());
+
+  const key = (req) => generateKey(req, "/waitTime", ["id"]);
+  router.get(
+    "/waitTime",
+    cache(key, redis),
+    asyncify(async (req, res) => {
+      try {
+        const query = await (req.query.id
+          ? db.waitTime(req.query.id)
+          : db.waitTime());
+
+        if (query) {
+          const data = JSON.stringify(query);
+
+          if (redis) {
+            redis.setex(key(req), 60, data);
+          }
+
+          res.status(200).send(data);
+        }
+        else {
+          res.status(400).send("No such facility id.");
+        }
+      } catch (err) {
+        console.log(err);
+        res.status(400).send(err);
+      }
```

Comment:
This was actually something I meant to suggest earlier in the semester.  
Some endpoints return something like this:
```
{success: false, error: err.message}
```
And others just return `err` or `err.message`. I think it'd be good to standardize it (at least in the new routes you write).  

Another thing: there are a bunch of places with the comment `// TODO Send actual error codes based on errors`. I didn't really understand what it meant but I think it might have to do with stuff like [status 404 vs. 400](https://www.reddit.com/r/rest/comments/cf173m/400_vs_404_for_nonexistent_entity/). In this case, I think status 404 makes more sense for "No such facility id." But again, these routes are all for internal use so it's probably not a big deal unless Flux decides to make a scalable public API.